controllers/cluster_controller.go

```diff
@@ -216,6 +217,20 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	}
 
+	// Delete the Cluster's Kubeconfig Secret, if any.
```

Comment:
If there is an OwnerReference, this shouldn't be needed.

Reply:
We don't set one today, should we?

Reply:
We should be adding the OwnerRef when we create, and if someone is providing a pre-existing kubeconfig secret, then we should likely not assume that we can delete it for them.

Reply:
Changed it up to add the owner ref